File: server/src/main/scala/kpn/api/common/tiles/ZoomLevel.scala
package kpn.api.common.tiles

object ZoomLevel {

  val bitmapTileMinZoom = 6
  val bitmapTileMaxZoom = 11

  val poiTileMinZoom = 11
  val poiTileMaxZoom = 15

  val vectorTileMinZoom = 12
  val vectorTileMaxZoom = 14
  val vectorTileMaxOverZoom = 20

  val newMinZoom: Int = 2
  val newMaxZoom: Int = 13

  val minZoom: Int = 2 // min(bitmapTileMinZoom, vectorTileMinZoom)
  val maxZoom: Int = 14 //max(bitmapTileMaxZoom, vectorTileMaxZoom)

  val all: Seq[Int] = ZoomLevel.minZoom.to(ZoomLevel.maxZoom)

  val nodeMinZoom = 12
}

File: client/src/app/components/ol/domain/zoom-level.ts
export class ZoomLevel {

  static readonly bitmapTileMinZoom = 6;
  static readonly bitmapTileMaxZoom = 11;

  static readonly vectorTileMinZoom = 12;
  static readonly vectorTileMaxZoom = 14;
  static readonly vectorTileMaxOverZoom = 20;

  static readonly poiTileMinZoom = 11;
  static readonly poiTileMaxZoom = 14;

  static readonly minZoom: number = Math.min(
    ZoomLevel.bitmapTileMinZoom,
    ZoomLevel.vectorTileMinZoom
  );

  static readonly maxZoom: number = Math.max(
    ZoomLevel.bitmapTileMaxZoom,
    ZoomLevel.vectorTileMaxZoom,
    ZoomLevel.vectorTileMaxOverZoom
  );

}
